blogp3.md

```diff
@@ -0,0 +1,434 @@
+### Creating the React App
+
+In the final part of these series, I'm going to be going over important steps in creating the frontend of our app in React. This tutorial won't cover the full code, so refer to [this](https://github.com/natalieh235/songrecproject/) github repo for all the deets.
+
+### Setting up the Client in React
+
+#### Step 1- Create the App
+
+If you haven't already, use the command `npm install create-react-app -g` to install create-react-app globally on your machine. Then, get back into your Visual Studio Code root directory and use these commands to make a React app:
+
+```
+create-react-app client
+cd client
+```
+
+You should now have two folders: server and client. 
+
+Now you can run these commands to install your dependencies and start running your app.
+
+```
+npm install
+npm start
+```
+
+Navigate to `http://localhost:3000` to see if your app is up and running. It should look something like this:
+
+[![startingreactpage](https://github.com/natalieh235/songrecproject/raw/blog/images/startingreactpage.png)](https://github.com/natalieh235/songrecproject/blob/blog/images/startingreactpage.png)
+
+#### Step 2- Create the Homepage
+
+So this React app will involve some very simple routing- basically this means that different urls will render different pages. For example, going to `localhost:3000/home` will take you to a homepage, while going to `localhost:3000/song` will perhaps show a list of songs. We're going to focus on creating a basic homepage, which will look something like this:
+
+[![reacthomepage](https://github.com/natalieh235/songrecproject/raw/blog/images/reacthomepage.png)](https://github.com/natalieh235/songrecproject/blob/blog/images/reacthomepage.png)
+
+Get started by making a new file called `Home.js`. I stuck this inside a folder `components`, since we're going to end up making a lot more files later on.
+
+Add in this code:
+
+```
+import React from 'react';
+
+function Home() {
+  return (
+    <div className="home-page">
+        <h1>Upload your face. We pick the songs.</h1>
+        <a href="https://localhost:8888">
+          <button className="green-btn">Login with Spotify</button>
+        </a>   
+    </div>
+  );
+}
+
+export default Home;
+```
+
+This is a simple functional component, and simply contains a header and button. Notice that the button is redirecting us to our OAuth server page.
+
+Return to your React app page and check to make sure the button takes you to the OAuth page, but don't log in quite yet.
+
+#### Step 2- Creating the Redirect Page
+
+After the user logs in, we need to redirect them to a different page in our React app. To do this, I'm creating a new file called `UploadFace.js`. Note: this name ends up being kind of misleading, since this ends being the page where all the uploading and song-finding happens, but just bear with me.
```

Comment:
```suggestion
After the user logs in, we need to redirect them to a different page in our React app. To do this, I'm creating a new file called `UploadFace.js`. Note: this name ends up being kind of misleading since this ends being the page where all the uploading and song-finding happens, but just bear with me.
```